ipaserver/install/dnskeysyncinstance.py

```diff
@@ -186,23 +187,57 @@ def __setup_softhsm(self):
                                'tokens_dir': paths.DNSSEC_TOKENS_DIR
                             }
         logger.debug("Creating new softhsm config file")
-        named_fd = open(paths.DNSSEC_SOFTHSM2_CONF, 'w')
-        named_fd.seek(0)
-        named_fd.truncate(0)
-        named_fd.write(softhsm_conf_txt)
-        named_fd.close()
-        os.chmod(paths.DNSSEC_SOFTHSM2_CONF, 0o644)
+        with open(paths.DNSSEC_SOFTHSM2_CONF, 'w') as f:
+            os.fchmod(f.fileno(), 0o644)
+            f.write(softhsm_conf_txt)
+
+        if constants.NAMED_SSL_ENGINE is not None:
+            # setup OpenSSL config file
+            # this one is needed because FreeIPA installation
+            # disables p11-kit-proxy PKCS11 module
+            # https://github.com/OpenSC/libp11
+            openssl_conf_txt = textwrap.dedent("""
+                # OpenSSL configuration file
```

Comment:
I don't like to have hard-coded configuration files hidden in the middle of lot of code. Please create a template in ``install/share/`` and use it here.

Reply:
That was the easiest way.
I don't insist, will be done in the next patch version.